python/ray/dag/compiled_dag_node.py

```diff
@@ -786,6 +848,17 @@ def _get_or_compile(
                         "or at least one other DAGNode as an input"
                     )
 
+        from ray.dag.constants import RAY_ADAG_ENABLE_DETECT_DEADLOCK
+
+        if RAY_ADAG_ENABLE_DETECT_DEADLOCK and not self._detect_deadlock():
```

Comment:
Yes, it shouldn't matter for users now that we are catching some more error cases properly in this PR, but I moved it because it was failing an assertion in _detect_deadlock. The problem was that the error check that is now above this block should have failed first, before we even tried to check for deadlock.